crates/wasmtime/src/store.rs

```diff
@@ -98,6 +98,7 @@ pub(crate) struct StoreInner {
     current_poll_cx: Cell<*mut Context<'static>>,
     out_of_gas_behavior: Cell<OutOfGas>,
     context_values: RefCell<HashMap<TypeId, Box<dyn Any>>>,
+    limiter: RefCell<Option<Arc<dyn wasmtime_runtime::ResourceLimiter>>>,
```

Comment:
It looks like this value doesn't change over time, so I don't think it needs to be backed by a `RefCell`?

Reply:
It currently does as it needs to sit on `inner` which is behind `Rc` and there's a circular reference between the proxy and the `Store`, which is why there is ugliness in `Store::new_with_limiter` to set this thing.

I'm not happy about it at all, so if there's a better way to do this such that the limiter can pass the store along in the callbacks, I'm definitely open to it.

Reply:
Hm ok, I had a question above about the `Store` argument and whether we can possibly remove that, but also here do you think it would make sense to remove the `Option` to effectively always have a limiting strategy?

Reply:
The limiter needs to be an `Option` in the runtime for both table and memory so that the pooling allocator is able to "take" the memory/table when deallocating the instance (swaps in `None`).

I don't think having it also `Option` in `Store` makes it that much worse and it does have the benefit of not boxing a new trait object every time a `Store` is created just to return `true` on memory/table grow operations.

If you're okay with it, I'd like to leave this one as-is.